woocommerce/wc-admin
# Move experimental list components to experimental package

Fixes #6928 

This PR moves the experimental list components (list, list item, collapsible list) to the experimental package. Exporting them as `List`, `ListItem`, and `CollapsibleList`.

### Detailed test instructions:

- Load up this branch and go to **WooCommerce > Home** having finished the onboarding flow
- The task list should be rendered correctly
- -  Create a 3PD task list by following the JS portion of [this tutorial](https://developer.woocommerce.com/2021/02/26/use-setup-tasks-to-provide-a-first-class-on-boarding-experience-for-merchants/), you can create a new extension using `npm run example -- --ext=add-task` 
Sample of filter:
```
addFilter(
	'woocommerce_admin_onboarding_task_list',
	'woocommerce_gpf',
	( tasks ) => {
		return [
			...tasks,
			{
				key: 'woocommerce_test_item_1',
				title: 'Title',
				container: <TestComp />,
				completed: true,
				visible: true,
				additionalInfo: 'additional info',
				time: '5 minutes',
				isDismissable: true,
                                 allowRemindMeLater: true
			}
                 ]
    }
);
```
- Check if the extendable task list renders fine, add more then 4 items, it should show the collapse option, this should also render fine.
- When running a `npm run clean && npm run dev` locally, it should build fine and **not** show a warning for the list item collapse export.
